masonry/src/widgets/virtual_scroll.rs

```diff
@@ -428,7 +428,7 @@ impl<W: Widget + FromDynWidget + ?Sized> VirtualScroll<W> {
         this.ctx.request_layout();
     }
 
-    fn post_scroll(&mut self, ctx: &mut EventCtx<'_>) {
+    fn post_scroll(&mut self, ctx: &mut UpdateCtx<'_>) {
         // We only lock scrolling if we're *exactly* at the end of the range, because
         // if the valid range has changed "during" an active scroll, we still want to handle
         // that scroll (specifically, in case it happens to scroll us back into the active
```

Comment:
It looks like you're only creating this new UserUpdate event so you can share the post_scroll function between two passes.

What I'd recommend doing instead is:

- Remove the `ctx` parameter for post_scroll.
- Have post_scroll return a bool (true is request_layout is needed, false otherwise).
- Move request_compose and request_layout to the callers.

Reply:
Then it could be a private struct or a newtype or an enum.